app/helpers/repository_helper.rb

```diff
@@ -168,8 +168,6 @@ def remove_folders(folders, user, repo, path: '/', txn: nil)
         path = File.join obj.path, obj.name
         if obj.is_a? Repository::RevisionFile
           files << path
-        else
-          dirs << path
```

Comment:
This will work well for the GitRepository. Can you double check that it works for subversion and memory repositories as well.